app/src/main/java/com/willowtree/vocable/settings/EditKeyboardFragment.kt

```diff
@@ -106,15 +117,22 @@ class EditKeyboardFragment : BaseFragment() {
                 if (!isDefaultTextVisible()) {
                     binding?.keyboardInput?.text?.let { text ->
                         if (text.isNotBlank()) {
-                            phrase.utterance = text.toString()
-                            viewModel.updatePhrase(phrase)
+                            phrase?.utterance = text.toString()
+                            phrase?.let {
+                                viewModel.updatePhrase(it)
+                            }
+                            viewModel.addNewPhrase(text.toString())
                         }
                     }
                 }
             }
         }
 
-        binding?.keyboardInput?.setText(phrase.utterance)
+        if (phrase == null) {
```

Comment:
You could use the elvis operator here to condense what's inside `setText` to `phrase?.utterance ?: R.string.keyboard_select_letters`

Reply:
so i get a weird error when i try to do this: 
`binding?.keyboardInput?.setText(phrase?.utterance ?: R.string.keyboard_select_letters)`

`None of the following functions can be called with the arguments supplied `

Reply:
We done resolved it - had to wrap `R.string.keyboard_select_letters` with `getString()` since the `setText` being called was expecting a string